packages/manager/modules/pci/src/components/project/storages/databases/text-accordion/directive.js

```diff
@@ -0,0 +1,13 @@
+import template from './template.html';
+
+export default /* @ngInject */ () => {
+  return {
+    restrict: 'E',
+    controller: 'textAccordionController',
+    controllerAs: 'textAccordionCtrl',
+    bindToController: {
+      value: '<',
+    },
+    template,
+  };
```

Comment:
Hi @JeremyDec,
This is more of a patch on top of an already existent table, This will depend on size of the text and change the datagrid column to accommodate really large texts (with a drop down to hide/show).
This can may be go as a change in datagrid but we need to take a call on that

Reply:
@varun257 I'm ok with having this change here, but my question is: Here you're declaring a directive, with a `restrict` property set to 'E', so this will make the directive to behave like a component. So the question: Why not declare here a component instead ? (Like you did for packages/manager/modules/pci/src/components/project/warning-message/warning.component.js)

Reply:
@JeremyDec, the purpose of this new element is more on the behaviour of the DOM element. And though it has `restrict: 'E'`,looking at the behaviour, this directive can also accept `restrict: 'A'` as well.